internal/pkg/encoding/json/schema/schema.go

```diff
@@ -139,6 +139,13 @@ func processErrors(errs []*jsonschema.ValidationError) error {
 		path = strings.ReplaceAll(path, "/", ".")
 		msg := strings.ReplaceAll(e.Message, `'`, `"`)
 		if strings.HasPrefix(e.AbsoluteKeywordLocation, "https://json-schema.org/") {
+			// Required field in a JSON schema should be an array of required nested fields.
+			// But, for historical reasons, in Keboola components, "required: true" is also used.
+			// In the UI, this causes the drop-down list to not have an empty value.
+			// For this reason, we can ignore the error.
+			if strings.HasSuffix(e.InstanceLocation, "/required") && e.Message == "expected array, but got boolean" {
+				continue
+			}
```

Comment:
V 2. commite som fixol samotny problem.

V niektorych schemach komponent mame `required: true`, aj ked field `required` ma by array, zoznam toho, ktore z nested fields su required.

`required: true` pouzivaju komponenty dost specificky, ak je to pouzite, tak drop-down list nema medzi moznostami empty string, napr. `yes/no`, namiesto `(empty)/yes/no`.

Standard o tom nic nehovori, bude to asi nejaka specialitka v UI json-editor libke

------------

Jedine riesenie, ktore pokryva vsetky moznosti, je ignorovat chybu, ktora vznikne z `required: true`.

Rozmyslali sme aj na konverzii `required bool` -> `required array ` + pridat to o level vyssie, ale nasli sme par situacii, kde by to nefungovalo.

Reply:
Alebo iny priklad.

Chceme do schemy pridat field `foo`, ktory bol predtym volitelny a teraz ma byt z neho enum.
Ak ho pridame, tak nam tam UI json-editor prida aj `(empty)` moznost.

No nemozeme dany field `foo` spravit required tak, ze by sme ho pridalo do nadradeneho `required array`.
To by nefungovala validacia starych konfiguracii, kde tento field nebol.

Teda KDS to riesi tak, ze danemu fieldu da `required: true` a nejak to v UI json-editor robi to, co chcu.

Je to trocha zamotane...